### PR TITLE
refactor: consume v2 Device SDK

### LINKED ISSUE #166: Consume V2 Device SDK (Virtual)
1. Update go.mod file to reference new v2.0.x version of module
2. Update all import paths for the module to include /v2 in the path.
3. Update Attribution.txt to add /v2 to the module reference.

See the PR for edgex-go for example:  https://github.com/edgexfoundry/edgex-go/pull/3032/files

### PR DESCRIPTION
**Pull request by hahattan**
Signed-off-by: Chris Hung <chris@iotechsys.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-virtual-go/blob/master/.github/CONTRIBUTING.md

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

### CODE REVIEW

File: cmd/res/configuration.toml
Comment (lenny-intel) [inline cmd/res/configuration.toml:57]: Note that this can be eliminated since the service doesn't have any secrets.

File: cmd/res/configuration.toml
Comment (lenny-intel) [inline cmd/res/configuration.toml:9]: Note that this can be eliminated since the service doesn't have any secrets. We have device-mqtt as an example of how these are used for services that have secrets.

File: cmd/res/device.virtual.binary.yaml
Comment (lenny-intel) [inline cmd/res/device.virtual.binary.yaml:1]: I don't think this is needed here? We recently removed apiVersion from the sub-DTOs. Same for all device profiles in this PR.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: cmd/res/device.virtual.binary.yaml
Comment (cloudxxx8) [inline cmd/res/device.virtual.binary.yaml:17]: Since it is the same as the Device Resource, we can totally eliminate deviceCommands in this profile

File: cmd/res/device.virtual.binary.yaml
Comment (cloudxxx8) [inline cmd/res/device.virtual.binary.yaml:11]: specify `isHidden: false` for all the normal Device Resource and Device Command because this is a sample profile

File: cmd/res/device.virtual.bool.yaml
Comment (cloudxxx8) [inline cmd/res/device.virtual.bool.yaml:43]: add one more resourceOperation:
    - { deviceResource: "EnableRandomization_Bool", defaultValue: "false" }

File: cmd/res/device.virtual.bool.yaml
Comment (cloudxxx8) [inline cmd/res/device.virtual.bool.yaml:24]: specify `isHidden: true` for all the "EnableRandomization_XXX"

File: cmd/res/device.virtual.bool.yaml
Comment (cloudxxx8) [inline cmd/res/device.virtual.bool.yaml:40]: Modify the name to "WriteBoolVaule" and apply to all the deviceCommands

File: cmd/res/device.virtual.bool.yaml
Comment (cloudxxx8) [inline cmd/res/device.virtual.bool.yaml:41]: modify to "W"

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

**Review by lenny-intel — COMMENTED** [review]
LGTM once @cloudxxx8 comments are addressed

**Review by cloudxxx8 — DISMISSED** [review]
LGTM

**Comment by cloudxxx8** [comment]
@lenny-intel need your approval to merge this PR

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**
